### ISSUE TITLE
Enhancement: Copy/paste workflow steps from one workflow to another

### ISSUE THREAD

**Issue by hepcat72** [galaxyproject/galaxy#8923]
Currently, you can copy an individual workflow step inside 1 workflow, but it would be really handy to be able to copy multiple steps from one workflow to another.  I envision this working in much the same way Node Red's workflow editor handles it: select multiple steps, copy (e.g. command-c), switch workflows, then paste.

This would be particularly useful when creating similar flows that use many of the same tools with the same parameters or when wanting to streamline multiple flows into one in order to share with a novice user who might have trouble identifying the outputs of flow 1 needed as inputs for flow 2.

**Comment by nsoranzo** [comment]
Have a look at subworkflows in the mean time, if you haven't yet.

**Comment by hepcat72** [comment]
Yeah, I have used them, although I don't want to run everything.  I would like to use just the relevant parts.  E.g. ChIP-Seq and CLIP-Seq use a lot of the same QC, but have different analyses.

**Comment by hepcat72** [comment]
I mean, I suppose I could duplicate the flow and remove the irrelevent bits, but I was wanting to come up with a full concise single workflow to share with other users.

**Comment by hepcat72** [comment]
I could also run both workflows on some data and then extract a workflow from the history.

**Comment by bgruening** [comment]
@hepcat72 you can copy single steps, there is an icon ontop of each box for that.

**Comment by hepcat72** [comment]
> @hepcat72 you can copy single steps, there is an icon ontop of each box for that.

How do you get it to copy to a separate workflow?  Does it put the json on the clipboard?

**Comment by hepcat72** [comment]
Just thought I would add a note about how [node-RED](https://nodered.org) does this for their workflows.  They have an export tool (selected from the top right menu) that allows you to choose to export selected nodes (i.e. tools/steps), an entire flow, or all flows.  It shows the json, which can be copied manually.  You can also just select multiple nodes and use a keyboard shortcut and it just copies the json to the clipboard.  Then you can paste it into any flow or use the corresponding import tool.
